initrd/etc/functions

```diff
@@ -81,6 +81,36 @@ confirm_totp()
 	echo
 }
 
+reseal_tpm_disk_decryption_key()
+{
+	TRACE "Under /etc/functions:reseal_tpm_disk_decryption_key"
+	#Can be called only on resealing, not on tpm reset:
+	#otherise primary handle and rollback would change.
+	if ! grep -q /boot /proc/mounts ; then
+		mount -o ro /boot \
+		|| recovery "Unable to mount /boot"
+	fi
+
+	if [ -s /boot/kexec_key_devices.txt ] || [ -s /boot/kexec_key_lvm.txt ]; then
+		warn "A TPM disk decryption key previoulsy sealed is now invalid since firmware measurements cannot be unsealed" 
+		warn "Renewing LUKS Disk Unlock Key to be unsealed by TPM Disk Unlock Key passphrase"
```

Comment:
This logic is pretty warning-heavy which I fear will cause people to become accustomed to ignoring warnings :thinking: 

Your suggestion to add 'info' is a good one IMO.  That can be a follow up if you want to stick to echo for now.

I'd at least suggest reducing 96, 101, 104, 106 from warning to echo - those aren't warnings, those are the actions being taken as a result of the warning condition.

I don't really think 95 is a warning either.  That's normal during a firmware update; it's a specifically designed behavior that the key can't be unsealed if the firmware measurements change.

I'm not sure 103 should be a warning.  It seems it doesn't really know what the problem is ("...provided?") so I'm not sure the warning is meaningful.  If there was a problem at this point preventing update_checksums from succeeding, you'd be stuck in a loop too.  Maybe update_checksums should warn if anything warning-worthy occurs?  Maybe it should die here rather than being stuck in a loop, or maybe have a finite number of attempts?

Reply:
@JonathonHall-Purism : hmmm.
I fixed in code warnings to be seperated from normal behaviors, being echo in the goal of not overwhelming the user with warnings which sleeps. I have not implemented info here, since that would require to review all echo in code to call info instead (and define a proper encapsulation which I have not thinked about yet: makes sense to have warning with ***, die with !!!, but what to use for info?) To be dealt subsequently.

>  It seems it doesn't really know what the problem is ("...provided?") so I'm not sure the warning is meaningful. If there was a problem at this point preventing update_checksums from succeeding, you'd be stuck in a loop too. Maybe update_checksums should warn if anything warning-worthy occurs? Maybe it should die here rather than being stuck in a loop, or maybe have a finite number of attempts?

Well as of current code in PR here, we run after TOTP/TPM reset, we generate checksums and signing fails at this point. We guide user into probable causes, which are signing problems, guiding the user into what could be the cause of failing. At this point, its either a timeout (no USER PIN error but timeout) or a bad User PIN provided. In both cases, gnupg will inform prior/after. We want to sign here, so limiting the number of attempts in loop doesn't make sense to me. Any other recommendation?

Reply:
Thanks, I think the changes to echo here help a lot in 8809588a.  Agree we can hold info for future work (but I don't think we _have_ to review all existing echoes as a blocker to add it, IMO it's OK if we get some high-value ones initially and review others over time.  Either way, future work.)

Re: infinite loop of update_checksums - I think you've convinced me that bailing after a finite number of tries would not be better.  I suppose if it is broken, we can probably Ctrl+C, possibly combined with Esc-Esc to bail from fbwhiptail/whiptail and some good timing, so at least there is a way out from a bad situation.

I think we can go with what you have to get some field experience with it and refine if we see issues.